### DISCUSSION TITLE
StyledProperty<Type> shortening

### DISCUSSION THREAD

**Discussion by jordi-z80** [AvaloniaUI/Avalonia#16030] [Q&A]
Hi,

I'm pretty new to Avalonia and when creating controls, I feel constantly repeating the following snippet

	public static readonly StyledProperty<string> Text1Property = AvaloniaProperty.Register<TheUserControl02, string> (nameof (Text1));

	public string Text1
	{
		get => GetValue (Text1Property);
		set => SetValue (Text1Property, value);
	}

Is this normal, or am I missing something? 

I'm saying this because it seems like the perfect candidate for the source code generator, but I still struggle finding how to do things in Avalonia, so it may already be there.

Consider it a suggestion if it's not there :)

**Comment by StefanKoell** [comment]
At least using Rider you can use a Live Snippet to quickly add styled properties. I think a source generator doesn't help here because the Register method has quite a few optional arguments with actions and stuff.

**Comment by thevortexcloud** [comment]
> Is this normal, or am I missing something?

it's normal, and WPF does the same thing (WPF docs [here](https://learn.microsoft.com/en-us/dotnet/desktop/wpf/properties/how-to-implement-a-dependency-property?view=netdesktop-8.0)). It's also just as annoying as it is there. There was someone who wrote a source generator to do it for you, but from memory there was some issue that made it not workable for some scenarios. I however can't recall what it was though.

**Comment by maxkatz6** [comment]
> I'm saying this because it seems like the perfect candidate for the source code generator

Once partial properties are possible, sure.

**Comment by stevemonaco** [comment]
There's https://github.com/jp2masa/Avalonia.PropertyGenerator as a sourcegen to experiment with. Haven't used it myself. Code snippets are good enough and I tuck away dependency properties in a MyControl.props.cs partial class file.
